Audio recording for macOS

Nephyx opened the pull request:
Added audio recording and playback for macOS using Apple's Objective-C API via pyobjus.
Audio files are recorded with 44.1 kHz sample rate in WAV format.

Closes issue #282.

tito commented on plyer/platforms/macosx/audio.py:
```diff
@@ -0,0 +1,81 @@
+        
+        if self._recorder:
+            self._recorder.record()
+            print("Recording...")
```
Please remove all the print in the implementation

tito commented on plyer/platforms/macosx/audio.py:
```diff
@@ -0,0 +1,81 @@
+
+class OSXAudio(Audio):
+    def __init__(self, file_path=None):
+        default_path = join(expanduser('~'), 'Desktop', 'audio.wav')
```
You should not pass a default value that point on nothing :) Should be None by default.

Nephyx replied:
Tried without passing a value of `None`, but I started getting the following error:
`>>> from plyer import audio
>>> audio.start()
Traceback (most recent call last):
  File "/Users/maroskovac/Dev/plyer/plyer/utils.py", line 96, in _ensure_obj
    obj = mod.instance()
  File "/Users/maroskovac/Dev/plyer/plyer/platforms/macosx/audio.py", line 82, in instance
    return OSXAudio()
TypeError: __init__() takes exactly 2 arguments (1 given)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/maroskovac/Dev/plyer/plyer/facades/audio.py", line 65, in start
    self._start()
  File "/Users/maroskovac/Dev/plyer/plyer/facades/audio.py", line 98, in _start
    raise NotImplementedError()
NotImplementedError`

tito commented on plyer/platforms/macosx/audio.py:
```diff
@@ -0,0 +1,81 @@
+        # of the last recorded audio file
+        self._player = AVAudioPlayer.alloc()
+        self._player = self._player.initWithContentsOfURL_error_(
+            self._current_file, None
```
It would be better to catch the error and raise an exception if something failed, rather than completely ignore.

Nephyx commented:
@tito here's a quick summary of my commits addressing your changes request:
- removed all prints
- kept passing `file_path` with default value of `None`, as removing it as per your suggestion causes `NotImplementedError` (more in my reply to your change request)
- added raising exceptions in case of recorder or player initialization errors (using only one NSError instance because of segmentation faults in case of multiple NSError allocs and actually, there's no need to have more than one instance)

tito commented:
Thanks you!
